main.ts

```diff
@@ -404,11 +404,28 @@ class DocumentRenderer {
 
 			const extension = this.getExtension(src);
 			if (extension === '' || extension === 'md') {
-				const file = this.getEmbeddedFile(src);
+				let [file_name, heading] = src.split('#');
+				const file = this.getEmbeddedFile(file_name);
 				if (file) {
 					// Not recursively rendering the embedded elements here. If someone turns up with a need for
 					// this it should be easy to adapt this.
-					const markdown = await this.app.vault.cachedRead(file);
+					let markdown = await this.app.vault.cachedRead(file);
+					// If embedding links to a heading, only add the embedded section
+					if (heading) { 
+						heading = heading.split(" ").join("[ \|]+")
```

Comment:
The logic for extracting a section from a markdown text should be done in its own function. You can move it to the generic lib functions section at the top.